### DISCUSSION TITLE
Loadtest: Abnormal disconnection problem.

### DISCUSSION THREAD

**Discussion by calmchang** [colyseus/colyseus#572] [General]
When using @colyseus/loadtest to test a remote server, it works fine with 10 simulated users. However, when simulating 30 or more users, the connection drops after several tens of seconds and the client disconnection code is 1006. How can I solve this issue?

**Comment by calmchang** [comment]
Adding to that, if players only join the room without sending any messages, they won't be disconnected automatically. However, if a message is sent at a frequency of 30ms, some players will be disconnected automatically

**Comment by calmchang** [comment]
mayCurrently, based on the test results, it seems that the disconnection issue is caused by the large data packets of Colyseus, which may be due to the handshakeTimeout of WebSocket. Is there any way to adjust this timeout value?

**Comment by endel** [comment]
Hi @calmchang, the loadtest tool is single-threaded and may struggle to keep up with high amount of clients+traffic, I'd suggest spawning more instances of the loadtest with a lower amount of connections for testing.

**Reply by calmchang** [reply]
After my tests, it seems that continuous sending of large data using websocket.send is causing memory overflow, and I currently don't know how to solve this issue.

**Comment by endel** [comment]
For documentation: https://github.com/colyseus/colyseus/issues/574

> In real-world scenarios, console.log may be used in onMessage or onStateChange to print messages. Such exceptions can easily mislead people into investigating in the wrong direction, thinking that there is an issue with the server.

The higher CPU usage is necessary for each connection in the load tester, the lower the number of connections it can keep up simultaneously. There's a need for parallelization on the load test tool, which is currently single-threaded.
